Modules/Filtering/ImageFilterBase/include/itkTernaryGeneratorImageFilter.h

```diff
@@ -0,0 +1,277 @@
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(TernaryGeneratorImageFilter);
+
+  /** Standard class type aliases. */
+  using Self = TernaryGeneratorImageFilter;
+  using Superclass = InPlaceImageFilter<TInputImage1, TOutputImage>;
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(TernaryGeneratorImageFilter, InPlaceImageFilter);
+
+  /** Some type alias. */
+  using Input1ImageType = TInputImage1;
+  using Input1ImagePointer = typename Input1ImageType::ConstPointer;
+  using Input1ImageRegionType = typename Input1ImageType::RegionType;
+  using Input1ImagePixelType = typename Input1ImageType::PixelType;
+  using DecoratedInput1ImagePixelType = SimpleDataObjectDecorator<Input1ImagePixelType>;
+
+  using Input2ImageType = TInputImage2;
+  using Input2ImagePointer = typename Input2ImageType::ConstPointer;
+  using Input2ImageRegionType = typename Input2ImageType::RegionType;
+  using Input2ImagePixelType = typename Input2ImageType::PixelType;
+  using DecoratedInput2ImagePixelType = SimpleDataObjectDecorator<Input2ImagePixelType>;
+
+  using Input3ImageType = TInputImage3;
+  using Input3ImagePointer = typename Input3ImageType::ConstPointer;
+  using Input3ImageRegionType = typename Input3ImageType::RegionType;
+  using Input3ImagePixelType = typename Input3ImageType::PixelType;
+  using DecoratedInput3ImagePixelType = SimpleDataObjectDecorator<Input3ImagePixelType>;
+
+  using OutputImageType = TOutputImage;
+  using OutputImagePointer = typename OutputImageType::Pointer;
+  using OutputImageRegionType = typename OutputImageType::RegionType;
+  using OutputImagePixelType = typename OutputImageType::PixelType;
+
+
+  using FunctionType = OutputImagePixelType (*)(const Input1ImagePixelType &,
+                                                const Input2ImagePixelType &,
+                                                const Input3ImagePixelType &);
+
+  using ConstRefFunctionType = OutputImagePixelType(const Input1ImagePixelType &,
+                                                    const Input2ImagePixelType &,
+                                                    const Input3ImagePixelType &);
+  using ValueFunctionType = OutputImagePixelType(Input1ImagePixelType, Input2ImagePixelType, Input3ImagePixelType);
+
+  /** Connect one of the operands for pixel-wise operation. */
+  void
+  SetInput1(const TInputImage1 * image1);
+  virtual void
+  SetInput1(const DecoratedInput1ImagePixelType * input1);
+  virtual void
+  SetInput1(const Input1ImagePixelType & input1);
+
+  /** Set the first operand as a constant. */
+  virtual void
+  SetConstant1(const Input1ImagePixelType & input1);
+
+  /** Get the constant value of the first operand. An exception is sent if
```

Comment:
sent -> raised (also below)

Reply:
I believe "thrown" would be the C++ terminology.